Core/Core/Router/Route.swift

```diff
@@ -51,6 +51,10 @@ public struct Route: Equatable {
         return Route("/courses/\(courseID)/assignments/\(assignmentID)/fileupload")
     }
 
+    public static func assignmentTextSubmission(courseID: String, assignmentID: String, userID: String) -> Route {
+        return Route("/courses/\(courseID)/assignments/\(assignmentID)/submissions/\(userID)/online_text_entry")
```

Comment:
I really like that you chose `online_text_entry` as the route to be consistent with the submission type. Nice touch!